lib/rules/no-whitespace-before-property.js

```diff
@@ -25,14 +46,37 @@ module.exports = function(context) {
             var prop = node.property;
 
             if (astUtils.isTokenOnSameLine(obj, prop)) {
-                if (sourceCode.isSpaceBetweenTokens(obj, prop)) {
-                    context.report({
-                        node: node,
-                        message: "Unexpected whitespace before property '{{ propName }}'.",
-                        data: {
-                            propName: prop.name
+                var tokenBeforeProp;
+                var tokenBeforePunc;
+
+                tokenBeforeProp = sourceCode.getTokenBefore(prop);
+                while (tokenBeforeProp.value !== "." && tokenBeforeProp.value !== "[") {
+                    tokenBeforeProp = sourceCode.getTokenBefore(tokenBeforeProp);
+                }
+
+                tokenBeforePunc = sourceCode.getTokenBefore(tokenBeforeProp);
+
+                if (tokenBeforeProp.value === ".") {
```

Comment:
Thank you for good work!

But I think this `if` statements are too complexity.
Others look good to me.

A code in my head:

```js
var leftToken, rightToken;
if (node.computed) {
    rightToken = findOpenBracket(node);                    // "["
    leftToken = sourceCode.getTokenBefore(rightToken);     // the last token of `node.object` or `)`
} else {
    rightToken = sourceCode.getFirstToken(node.property);  // the token of `node.property`
    leftToken = sourceCode.getTokenBefore(rightToken, 1);  // the last token of `node.object` or `)`
}

if (sourceCode.isSpaceBetweenTokens(leftToken, rightToken)) {
    reportError(node, rightToken); // IMO, in computed properties case, to show "before '['" is more understandable.
}
```